File: Deepspace_Java/P2/src/deepspace/Loot.java
/**
 * @author BOLÍVAR PELÁEZ, CLARA
 * @author RUIZ GÓMEZ, SOLEDAD
 * 2º DGIIM - PDOO - Practice 1 DeepSpace
 */
package deepspace;

/**
 * @title Loot
 * @brief It represents the loot that a station can get if it beats an enemy starship
 */
public class Loot {
    
    private int nSupplies;
    private int nWeapons;
    private int nShields;
    private int nHangars;
    private int nMedals;
    
    
    /**
    * @brief Constructor
    */
    Loot (int s, int w, int sh, int h, int m) {
        this.nSupplies = s;
        this.nWeapons  = w;
        this.nShields  = sh;
        this.nHangars  = h;
        this.nMedals   = m;
    }
    
    /**
    * @brief Getter with package visibility
    * @return nSupplies value
    */
    int getNSupplies () {
        return nSupplies;
    }
    
    /**
    * @brief Getter with package visibility
    * @return nWeapons value
    */
    int getNWeapons () {
        return nWeapons;
    }
    
    /**
    * @brief Getter with package visibility
    * @return nShields value
    */
    int getNShields () {
        return nShields;
    }
    
    /**
    * @brief Getter with package visibility
    * @return nHangars value
    */
    int getNHangars () {
        return nHangars;
    }
    
    /**
    * @brief Getter with package visibility
    * @return nMedals value
    */
    int getNMedals () {
        return nMedals;
    }
    
    /**
    * @brief toString
    * @return String with info about the instance
    */
    public String toString () {
        
        String output = "Loot [ nSupplies " + nSupplies 
                        + " ; nWeapons " + nWeapons 
                        + " ; nShields " + nShields 
                        + " ; nHangars " + nHangars 
                        + " ; nMedals " + nMedals + " ]";
        
        return output;
    }
 
}

File: Deepspace_Ruby/P1/Loot.rb
# BOLÍVAR PELÁEZ, CLARA
# RUIZ GÓMEZ, SOLEDAD
# 2º DGIIM - PDOO - Practice 1 DeepSpace

# Loot
# It represents the loot that a station can get if it beats an enemy starship

module DeepSpace

        class Loot 

            # Constructor
            def initialize (s, w, sh, h, m)
                @nSupplies  = s
                @nWeapons   = w 
                @nShields   = sh 
                @nHangars   = h
                @nMedals    = m 
            end

            # Getters
            def nSupplies
                @nSupplies
            end

            def nWeapons
                @nWeapons
            end

            def nShields
                @nShields
            end

            def nHangars
                @nHangars
            end

            def nMedals
                @nMedals
            end
        end

end